package/MDAnalysis/analysis/density.py

```diff
@@ -394,8 +395,18 @@ def _prepare(self):
                 logger.warning(msg)
             # Generate a copy of smin/smax from coords to later check if the
             # defined box might be too small for the selection
-            smin = np.min(coord, axis=0)
-            smax = np.max(coord, axis=0)
+            try:
+                smin = np.min(coord, axis=0)
+                smax = np.max(coord, axis=0)
+            except ValueError as err:
+                msg = ("No atoms in AtomGroup at input time frame. "
+                       "Grid for density could not be automatically"
+                       " generated. A user defined grid was found"
+                       " and is being used as the density grid")
```

Comment:
My example message for the other warning can't be directly copied here, the context is different. It's incorrect to say that you were attempting to "automatically generate" the grid since it was manually set by the user. Please amend this to better fit the context of the error.

Reply:
`msg = ("No atoms in AtomGroup at input time frame. "`
                 `           "Density will contain a grid constructed with`
                 `           "user defined values ")`

Reply:
I think it's worth considering why we are emitting the warning. Is it reasonable to be warning folks that we're following expected operation (i.e. they asked to use a user defined grid, is it informative to have us tell them we're doing this)?

Rather than thinking about what the code is doing next, it's best to phrase the warning in terms of the consequences of not having any atoms in the atomgroup. What does this mean in terms of the user grid? How does it impact us checking if the user grid is large enough to accommodate the selected atoms?

Reply:
Hmm, that would mean that the user should have made sure that they enter a grid large enough to accomodate atoms in the next frames.

Reply:
`msg = ("No atoms in AtomGroup at input time frame. "`
             `           "Please ensure grid values are in a large range to`
             `           "accommodate all selected atoms ")`

Reply:
I'm not sure "large range" is the right set of words here. Maybe something like "This be may be intended; please ensure that your grid selection covers the atomic positions you wish to capture."?